File: Data Structures & Algorithms/majority-element/submission-3.java
class Solution {
    public int majorityElement(int[] nums) {
        int count=0;
        int ele=nums[0];
        for(int i=0;i<nums.length;i++){
            if(count==0 && ele!=nums[i]){
                ele=nums[i];
            }
            else if(ele!=nums[i])
                count--;
            else
                count++;
        }
        return ele;
    }
}

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-3.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode[]} lists
     * @return {ListNode}
     */
    mergeKLists(lists) {
        const minHeap = new MinPriorityQueue((ele)=> ele.val);
        for(const head of lists){
            if(head!== null)
                minHeap.enqueue(head);
        }
        const dummy = new ListNode();
        let curr = dummy;
        while(!minHeap.isEmpty()){
            let node = minHeap.dequeue();
            curr.next = node;
            curr= curr.next;
            if(node.next!==null)
                minHeap.enqueue(node.next);
            
        }
        return dummy.next;
    }
}
